soc/arm/raspberrypi/rp2/pinctrl_soc.h

```diff
@@ -0,0 +1,66 @@
+#ifndef __RP2_PINCTRL_SOC_H__
+#define __RP2_PINCTRL_SOC_H__
+
+#include <dt-bindings/pinctrl/rp2040-pinctrl.h>
+
+/**
+ * @brief Type to hold a pin's pinctrl configuration.
+ */
+typedef struct {
+	/** Pin number 0..29 */
+	uint32_t pin_num;
+	/** Alternative function (UART, SPI, etc.) */
+	uint32_t alt_func;
+	/** Maximum current used by a pin, in mA */
+	uint32_t drive_strength;
+	/** Slew rate, may be either false (slow) or true (fast) */
+	bool slew_rate_fast;
+	/** Enable the internal pull up resistor */
+	bool pullup;
+	/** Enable the internal pull down resistor */
+	bool pulldown;
+	/** Enable the pin as an input */
+	bool input_enable;
+	/** Enable the internal schmitt trigger */
+	bool schmitt_enable;
+} pinctrl_soc_pin_t;
```

Comment:
please, optimize this struct (potentially using a bit field). This data goes in ROM for every pin, so it's worth doing. You can take existing implementations as a reference.